employee/read_only.c

```diff
@@ -0,0 +1,44 @@
+/* This program is very similar to "full_privileges.c"
+ * but here we set the permissions of the capability
+ * to LOAD, i.e. read-only. So the user is not allowed to
+ * change the salary.
+ */
+
+#include "../include/common.h"
+#include "include/employee.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+	struct employee employee = {0, "Good", "Employee", 55000};
+	double salary = 0.0;
+	// set the permissions to read-only
+	struct employee *ro_employee = set_read_only(&employee);
+	inspect_pointer(ro_employee);
+	print_details(ro_employee);
+	printf("The struct is read-only so trying to change the salary will make the program crash...");
+	fflush(stdout);
```

Comment:
`printf(); fflush(); printf(); fflush` is equivalent to `printf(); fflush` with a "\n" in the printf string.

Reply:
Sometimes, cheri does weird thing. It was just part of the debugging. I will remove the extra statements.

Reply:
I don't think we can pretend that CHERI can have done anything weird here ;)

Reply:
Fixed in 0843e79.